src/snapshot.c

```diff
@@ -22,24 +22,25 @@ static int generate_snapshot(const char *file_name, uint8_t *buf, int buf_size)
         return 1;
     }
 
-    fwrite("/* This file was auto-generated */\n\n", 1, 36, f);
-    fwrite("#include \"zjs_common.h\"\n\n", 1, 25, f);
-    fwrite("const uint32_t snapshot_bytecode[] = {\n", 1, 38, f);
+    size_t bs = sizeof(char);
+    fwrite("/* This file was auto-generated */\n\n", bs, 36, f);
+    fwrite("#include \"zjs_common.h\"\n\n", bs, 25, f);
+    fwrite("const uint32_t snapshot_bytecode[] = {\n", bs, 38, f);
 
+    char byte[11];
     for (int i = 0; i < buf_size; i++)
     {
         if (i > 0) {
-            fwrite(",", 1, 1, f);
+            fwrite(",", bs, 1, f);
         }
-        char byte[5];
-        snprintf(byte, 5, "0x%02x", snapshot_buf[i]);
-        fwrite(byte, 1, 4, f);
+        snprintf(byte, 11, "0x%08lx", (unsigned long)buf[i]);
```

Comment:
Remove the l and make this (unsigned int) just for consistency with what I've been doing elsewhere.

And yeah I'd prefer to just leave the 'bs' var out, it's confusing I think.